### PR TITLE
Get conductor voltage

### PR DESCRIPTION
**Pull request by roelof-groenewald**
Added functionality from the Python side to evaluate a time dependent expression using the WarpX parser. This is now used in the direct solver to evaluate the boundary potentials. Thereby the direct solver test (`test_poisson_pseudo_1d.py`) already functions as a test for this new functionality.

### CODE REVIEW

File: mewarpx/mewarpx/mwxrun.py [hunk 1]
```diff
@@ -250,5 +251,22 @@ def set_phi_grid(self, phi_data):
...
+            expr (str or float): Expression to evaluate.
+
+        Returns:
+            (float) Value of the expression at the current simulation time.
+        """
+        if isinstance(expr, float):
+            return expr
+        else:
+            return _libwarpx.libwarpx.eval_expression_t(
+                ctypes.c_char_p(expr.encode('utf-8'))
```
Comment (nruof) [inline mewarpx/mewarpx/mwxrun.py]: This line is throwing an attribute error in circle ci, when running the test_circle_emitter.py

AttributeError: 'int' object has no attribute 'encode'

**Review by nruof — CHANGES_REQUESTED** [review]

**Review by EtherealMC-Bit — APPROVED** [review]
LGTM

File: Source/Python/WarpXWrappers.cpp [hunk 2]
```diff
@@ -567,4 +567,13 @@ extern "C"
...
         mypc.Redistribute();
     }
+
+    amrex::Real eval_expression_t ( const char* char_expr ) {
+        WarpX& warpx = WarpX::GetInstance();
+        const std::string expr(char_expr);
+
+        auto parser = makeParser(expr, {"t"});
+        auto parser_exe = parser.compileHost<1>();
+        return parser_exe(warpx.gett_new(0));
```
Comment (peterscherpelz) [inline Source/Python/WarpXWrappers.cpp]: I would add a comment or something that this forces level=0, OR add level as an input from python...

File: mewarpx/mewarpx/mwxrun.py (hunk 1 above)
Comment (peterscherpelz) [reply mewarpx/mewarpx/mwxrun.py]: Looks like you can check the abstract base class https://docs.python.org/3/library/numbers.html#numbers.Real or, just check it's not a string - actually I like that best, suggestion as separate comment.

Comment (peterscherpelz) [reply mewarpx/mewarpx/mwxrun.py]:
```
        if isinstance(expr, str):
            return _libwarpx.libwarpx.eval_expression_t(
                ctypes.c_char_p(expr.encode('utf-8'))
        else:
            return expr
```

**Review by peterscherpelz — COMMENTED** [review]
Good stuff, few minor things.

**Review by peterscherpelz — APPROVED** [review]
Good stuff!

**Pull request merged**